src/client/sandbox/event/selection.js

```diff
@@ -117,27 +135,30 @@ export default class Selection {
     }
 
     getSelection (el) {
-        var changeType      = Selection._needChangeInputType(el);
-        var activeElement   = domUtils.getActiveElement(domUtils.findDocument(el));
-        var isElementActive = activeElement === el;
-        var savedType       = el.type;
-        var selection       = null;
+        var changeType           = Selection._needChangeInputType(el);
+        var useInternalSelection = Selection._needForInternalSelection();
+        var activeElement        = domUtils.getActiveElement(domUtils.findDocument(el));
+        var isElementActive      = activeElement === el;
+        var savedType            = el.type;
+        var selection            = null;
 
         // HACK: (A problem with input type = ‘number’ after Chrome is updated to v.33.0.1750.117 and in
         // Firefox 29.0. T101195) To get selection, if the input type is  'number' or 'email', we need to change
         // the type to text (B254340). However, the type is changed asynchronously in this case. To force type changing,
         // we need to call blur.Then call focus to make the element active.
         if (changeType) {
-            if (isElementActive)
+            if (!useInternalSelection && isElementActive)
                 this.focusBlurSandbox.blur(el, null, true);
 
             el.type = 'text';
         }
 
+        var internalSelection = el[INTERNAL_PROPS.selectionProperty];
+
         selection = {
-            start:     el.selectionStart,
-            end:       el.selectionEnd,
-            direction: el.selectionDirection
+            start:     internalSelection ? internalSelection.selectionStart : el.selectionStart,
```

Comment:
if we didn't set internal selection to this element before it will return `{ null, null, null }`, but should some default values. I told you before about it